pkg/controller/controller_binding.go

```
@@ -349,38 +353,75 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for creating/updating the Secret
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
```

So, I was thinking about this last night, and I wonder if we can just return an error here and let the controller retry this entire thing - that way we get retries / backoff / etc without increasing the complexity of this method.  According to the spec, it should be possible to re-perform the bind and get the same credentials back.